apps/api/schedule.py

```diff
@@ -169,6 +170,56 @@ def post(self, proposal_type):
         return [t.id for t in res]
 
 
+class C3VOCPublishingWebhook(Resource):
+    method_decorators = {"post": [_require_video_api_key]}
+
+    def post(self):
+        if not request.is_json:
+            abort(415)
+
+        payload = request.get_json()
+
+        try:
+            conference = payload["fahrplan"]["conference"]
+            proposal_id = payload["fahrplan"]["id"]
+        except KeyError:
+            abort(422)
+
+        if not payload["is_master"]:
+            # c3voc *should* only send us information about the master
+            # encoding getting published. Aborting early ensures we don't
+            # accidentially delete video information from the database.
+            abort(406)
+
+        if conference != f"emf{event_year()}":
+            abort(422)
+
+        proposal = Proposal.query.get_or_404(proposal_id)
+
+        if payload["voctoweb"]["enabled"] and payload["voctoweb"]["frontend_url"]:
+            proposal.c3voc_url = payload["voctoweb"]["frontend_url"]
+            proposal.video_recording_lost = False
+        else:
+            # This allows c3voc to notify us if videos got depublished
+            # as well. We do not explicitely set 'video_recording_lost'
+            # here because the video might only need fixing audio or
+            # such.
+            proposal.c3voc_url = ""
```

Comment:
I'm wondering about the use-cases for this: is this also necessary to handle situations where the content should be removed?  (some Twitter/X API events for example include events to indicate that posts should be removed downstream)

Reply:
Sometimes we (c3voc) we get requests to take a video down for various reasons (accidentially published, copyright/slide fixes or such things). In this case the reference to the no-longer-existing videos should be removed from emf website as well, because we don't want to link to "this video has been removed" pages.

Reply:
Ok, thanks - that makes sense; and I suppose that, to be cautious, we would also want to remove the `youtube_url` when that happens - a slight conflict with my previous suggestion not to edit existing values for that attribute.

Reply:
i have reverted the "don't update existing youtube urls" commit. I'd wait for input of @wlcx on this as well.